utils/time.js

```diff
@@ -0,0 +1,24 @@
+const hrTimeSupport = typeof process !== 'undefined' && process.hrtime;
+
+// since hrtime isn't available, we can ignore the input parameters
+function nowLegacy() {
+  return Date.now() * 1000;
+}
+
+function nowHrTime(startTimestamp, startTick) {
+  if (startTimestamp && startTick) {
+    const hrtime = process.hrtime(startTick);
+    const elapsedMicros = Math.floor(hrtime[0] * 1000000 + hrtime[1] / 1000);
+    return startTimestamp + elapsedMicros;
+  } else {
+    return Date.now() * 1000;
+  }
+}
+
+// Returns the current time in epoch microseconds
+// if startTimestamp and startTick are present, process.hrtime is used
+// See https://nodejs.org/api/process.html#process_process_hrtime_time
+module.exports.now = hrTimeSupport ? nowHrTime : nowLegacy;
```

Comment:
If we check for `hrTimeSupport` in `nowHRTime` then we don't need to write `Date.now() * 1000;` in two places (and we don't need `nowLegacy` - which is a little oxymoronic anyway).

```js
const now = (startTimestamp, startTick) => {
  if (!hrTimeSupport || !startTimestamp || !startTick) {
    return Date.now() * 1000;
  }
  // ...your code without the if / else
};

module.exports = {
  now,
  hrtime: hrTimeSupport ? process.hrtime : () => undefined,
};
```

Reply:
good call